main.go

```diff
@@ -472,6 +472,23 @@ func makeContextWithTrace(ctx context.Context, t transport.Transport, request *t
 
 func makeInitialRequest(out output, transport transport.Transport, serializer encoding.Serializer, req *transport.Request) {
 	response, err := makeRequestWithTracePriority(transport, req, 1)
+
+	var bs []byte
+	outSerialized := map[string]interface{}{}
+	if response != nil && len(response.Headers) > 0 {
+		outSerialized["headers"] = response.Headers
+	}
+	defer func() {
```

Comment:
nit: even if previously we had this piece of code in the function `makeInitialRequest`, can we expose in another method for better readability? For instance `defer printOutputInJSON(outSerialized)`